replica-common/src/main/java/com/graphhopper/OsmHelper.java

```diff
@@ -80,7 +84,12 @@ public static Map<String, String> parseTagsFromOsmElement(ReaderElement wayOrRel
             if (OSM_NAME_TAG.equals(tag)) {
                 // Parse street name, which is a concat of `name` and `ref` tags (if present)
                 parsedTagValues.put(tag, getConcatNameFromOsmElement(wayOrRelation));
-            } else {
+            } else if (OSM_RELATION_ID.equals(tag)) {  // special case, OSM_RELATION_ID isn't a real tag
+                parsedTagValues.put(tag, Long.toString(wayOrRelation.getId()));
+            } else if (OSM_RELATION_NAME.equals(tag)) {  // special case, OSM_RELATION_NAME isn't a real tag
+                parsedTagValues.put(tag, getTagValueFromOsmElement(wayOrRelation, "name"));
+            }
```

Comment:
nit: since these two special cases are specific to relations and aren't appropriate for ways, maybe it'd be a little cleaner to put it in the relation-specific part of `updateOsmIdToWayTags` instead of this shared method? and if that complicates `ALL_RELATION_TAGS_TO_PARSE`, we could just remove that constant since the placeholders require diff logic from the formal relation tags. up to you!

Reply:
just to be sure, what's the relation-specific part of `updateOsmIdToWayTags` you're referring to here? Is it the spot right before the call to that method [here](https://github.com/replicahq/graphhopper/pull/162/files/6bd337ff3dbab005449daa933a0783cc684f1e1d#diff-221bd148393011a56e5bef015d107b765efac49201a7cb43ac9819f3702ce0fcR162) maybe?

My intention with these two methods was for `parseTagsFromOsmElement` to be a universal method for both ways + relations to parse tag data, including special cases - the fact that they're both subclasses of `ReaderElement` makes that handy. `updateOsmIdToWayTags` right now is set up to take those already-parsed values and update the main map where we're storing the "final" values, and isn't concerned with where those values were parsed from

Reply:
yeah that area you linked is what i had in mind! and yeah, makes sense about wanting a universal parsing method. part of what caught my eye is that it would be pretty weird to pass this method an osm way and `OSM_RELATION_ID`/`OSM_RELATION_NAME` as tags to parse (realize that doesn't happen currently, but the method would allow it). these placeholder tags are relation-specific, so we could add an `instanceof` check or something here, but using `instanceof` in java is always a little smelly. and given we need special case logic for the placeholder tags anyways, maybe it's clearer to separate that into a relation-specific part of the code. anyways this is minor and non-blocking, i don't mean for this to be a distraction!

Reply:
Ahh I see what you're saying! I agree that the way this is set up is a bit too forgiving of mixing way/relation types with tags that don't apply to that OSM element type.

I chose to make a slightly different refactor to address the issue, keeping the same universal parsing method but putting it behind two new methods (one for Ways, one for Relations). The Ways method is hardcoded to always use `ALL_WAY_TAGS_TO_PARSE`, while the Relation method still takes a set of tags to parse as input. The more specific (subclass) input types of these new methods ensure you'll never try to parse relation-only tags from a way, but still allow flexibility when parsing tags from a relation